{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "FILELIST = 'C:/Users/Brian/Desktop/Boston University/Databases/Group_Project/FASTAs/distincts.csv'\n",
    "\n",
    "import pandas\n",
    "import csv\n",
    "\n",
    "filenames = list(pandas.read_csv(FILELIST)['PDB ID'])\n",
    "genenames = list(pandas.read_csv(FILELIST)['gene name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "OUTFILE = 'C:/Users/Brian/Desktop/Boston University/Databases/Group_Project/FASTAs/fastas2.txt'\n",
    "\n",
    "for i in range(len(filenames)):\n",
    "    FASTAFILE = 'C:/Users/Brian/Desktop/Boston University/Databases/Group_Project/FASTAs/Group_Project-master/Group_Project-master/src/media/fastas/'+filenames[i]+'.fasta'\n",
    "    with open(FASTAFILE) as csv_file:\n",
    "        csv_reader = csv.reader(csv_file, delimiter=',')\n",
    "        seq = next(csv_reader)\n",
    "        \n",
    "    csv_file.close()\n",
    "    \n",
    "    with open(OUTFILE, 'a') as writeFile:\n",
    "        writer = csv.writer(writeFile)\n",
    "        hdr = '>'+genenames[i]\n",
    "        writer.writerows([[hdr]])\n",
    "        writer.writerows([seq])\n",
    "        writer.writerows('')\n",
    "\n",
    "    writeFile.close()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "April 18, 2019:\n",
    "\n",
    "Have written Python scripts that can \n",
    "\n",
    "1) pull together multiple FASTA files into a single text file\n",
    "2) pass that text file to Clustal Omega and get a multiple-alignment file back\n",
    "3) pass that MSA file to simple phylogeny (also at EMBL) and get a Newick tree back\n",
    "\n",
    "1) is \"Make_single_FASTA_file\"\n",
    "    * An issue: for the phylogeny, we'll actually only want one FASTA for each distinct protein \n",
    "    (we have multiple instances of many proteins); will need a script, perhaps pymysql, that pulls a\n",
    "    list of PDB IDs that have *distinct* UniProt IDs, and then passes that list to \"Make_single_FASTA_file\".\n",
    "    \n",
    "2) is \"clustalo.py\", in Group_Project/Group_Project/src. Note that it's the code available from EMBL, but\n",
    "with a couple of modifications: it imports ssl, and then invokes an option whenever urlopen is called that \n",
    "prevents urlopen from asking for a security certificate. This is important because urlopen was triggering errors\n",
    "based on security-certificate issues.\n",
    "    Syntax for calling it from cmd: python clustalo.py --email spitzer@bu.edu --stype protein --sequence fastas.txt\n",
    "\n",
    "3) is \"simple_phylogeny.py\". Same location, same modifications.\n",
    "\n",
    "A couple of issues outstanding:\n",
    "\n",
    "* When EMBL returns a file (the .clustal_num file from Clustal Omega, and the Newick tree from simple phylogeny,\n",
    "which is a .phylotree.ph file), it assigns an arbitrary name to the file; the next script in the pipeline will \n",
    "somehow need to know the name of that file in order to pass it to the next script.\n",
    "\n",
    "* Need to figure out how to pass the Newick file to jsphylo\n",
    "\n",
    "* Also need to figure out how to embed MSAViewer and pass the clustal_num file to it.\n",
    "\n",
    "* Write code that automatically writes code for jsphylo. Need to ask Amanda about what names the tips of the tree\n",
    "should have (i.e., what are users going to be browsing for? Gene name? Protein name?), and then determine what clicking\n",
    "on the tips should do.\n",
    "\n",
    "* Need to link everything together into a single pipeline.\n",
    "\n",
    "\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['5-hydroxytryptamine receptor 1B',\n",
       " '5-hydroxytryptamine receptor 2B',\n",
       " '5-hydroxytryptamine receptor 2C',\n",
       " 'Adenosine receptor A1',\n",
       " 'Adenosine receptor A2a',\n",
       " 'Apelin receptor',\n",
       " 'Beta-1 adrenergic receptor',\n",
       " 'Beta-2 adrenergic receptor',\n",
       " 'C5a anaphylatoxin chemotactic receptor 1',\n",
       " 'Calcitonin gene-related peptide type 1 receptor',\n",
       " 'Calcitonin receptor',\n",
       " 'Cannabinoid receptor 1',\n",
       " 'C-C chemokine receptor type 2',\n",
       " 'C-C chemokine receptor type 5',\n",
       " 'C-C chemokine receptor type 9',\n",
       " 'Corticotropin-releasing factor receptor 1',\n",
       " 'C-X-C chemokine receptor type 1',\n",
       " 'C-X-C chemokine receptor type 4',\n",
       " 'D(2) dopamine receptor',\n",
       " 'D(3) dopamine receptor',\n",
       " 'D(4) dopamine receptor',\n",
       " 'Delta-type opioid receptor',\n",
       " 'Endothelin receptor type B',\n",
       " 'Free fatty acid receptor 1',\n",
       " 'Frizzled-4',\n",
       " 'Glucagon receptor',\n",
       " 'Glucagon-like peptide 1 receptor',\n",
       " 'G-protein coupled receptor homolog US28',\n",
       " 'Histamine H1 receptor',\n",
       " 'Kappa-type opioid receptor',\n",
       " 'Lysophosphatidic acid receptor 1',\n",
       " 'Metabotropic glutamate receptor 1',\n",
       " 'Metabotropic glutamate receptor 5',\n",
       " 'Muscarinic acetylcholine receptor M1',\n",
       " 'Muscarinic acetylcholine receptor M2',\n",
       " 'Muscarinic acetylcholine receptor M3',\n",
       " 'Muscarinic acetylcholine receptor M4',\n",
       " 'Mu-type opioid receptor',\n",
       " 'Neuropeptide Y receptor type 1',\n",
       " 'Neurotensin receptor type 1',\n",
       " 'Nociceptin receptor',\n",
       " 'Orexin receptor type 1',\n",
       " 'Orexin receptor type 2',\n",
       " 'P2Y purinoceptor 1',\n",
       " 'P2Y purinoceptor 12',\n",
       " 'Parathyroid hormone-parathyroid hormone-related peptide receptor',\n",
       " 'Platelet-activating factor receptor',\n",
       " 'Prostaglandin D2 receptor 2',\n",
       " 'Prostaglandin E2 receptor EP3 subtype',\n",
       " 'Prostaglandin E2 receptor EP4 subtype',\n",
       " 'Proteinase-activated receptor 1',\n",
       " 'Proteinase-activated receptor 2',\n",
       " 'Rhodopsin',\n",
       " 'Smoothened homolog',\n",
       " 'Sphingosine 1-phosphate receptor 1',\n",
       " 'Substance-P receptor',\n",
       " 'Thromboxane A2 receptor',\n",
       " 'Type-1 angiotensin II receptor',\n",
       " 'Type-2 angiotensin II receptor']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genenames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
